### PR TITLE
Add Server-Timing header

### PR DESCRIPTION
**Pull request by owais**
## Why

To enable trace linkage between the APM and RUM products.

## What

Updated instrumentation to return Server-Timing header with the value being the trace context.

### CODE REVIEW

File: flask_opentracing/tracing.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def __init__(
         app=None,
         traced_attributes=[],
         start_span_cb=None,
+        trace_response_header=False,
```
Comment (pellared) [inline flask_opentracing/tracing.py]: Do we want to have it disabled by default?
I am supposed to enable it by default in https://github.com/signalfx/splunk-otel-go per https://github.com/signalfx/splunk-otel-go/pull/18#issuecomment-808338996

File: flask_opentracing/tracing.py [hunk 2]
```diff
@@ -162,6 +164,22 @@ def _after_request_fn(self, response=None, error=None):
...
             scope.span.set_tag(tags.HTTP_STATUS_CODE, response.status_code)
+
+            if self._trace_response_header:
+                expose_headers = response.headers.get(
+                    "Access-Control-Expose-Headers", ""
+                )
+                if expose_headers:
+                    expose_headers += ","
+                expose_headers += "Server-Timing"
+                response.headers["Access-Control-Expose-Headers"] = expose_headers
```
Comment (pellared) [inline flask_opentracing/tracing.py]: I think it is worth to add an assertion if unit test to ensure that `Server-Timing` is added to `Access-Control-Expose-Headers` HTTP Header. Especially that there is logic here with the comma separation, 

Side comment. @johnbley have you checked that this is how `http.Header.Add` works in Go? I am not sure if I should double-check this implementation https://github.com/signalfx/signalfx-go-tracing/pull/34 or is it fine.

File: tests/test_flask_tracing.py [hunk 3]
```diff
@@ -13,7 +13,9 @@
 test_app = app.test_client()
 
 
-tracing_all = FlaskTracing(MockTracer(FlaskScopeManager()), True, app, ["url"])
+tracing_all = FlaskTracing(
+    MockTracer(FlaskScopeManager()), True, app, ["url"], None, True
```
Comment (pellared) [inline tests/test_flask_tracing.py]: this constructor gets ugly.
any pattern in Python to make it more maintainable/readable?
some config object?

**Review by pellared — COMMENTED** [review]
Added some comments. It would be good to make the PR description better. Also, we may consider adding a Changelog.

File: flask_opentracing/tracing.py (hunk 1 above)
Comment (owais) [reply flask_opentracing/tracing.py]: Java docs say it is disabled by default. I copied that.

Comment (owais) [reply flask_opentracing/tracing.py]: Enabling by default makes sense to me though. Will do that and confirm internally.

File: tests/test_flask_tracing.py (hunk 3 above)
Comment (owais) [reply tests/test_flask_tracing.py]: Yea, it'd be better to use named params to make it more readable and obvious.

File: flask_opentracing/tracing.py (hunk 1 above)
Comment (pellared) [reply flask_opentracing/tracing.py]: OK. Keep me in the loop as if it should not be enabled by default then https://github.com/signalfx/splunk-otel-go/pull/18#issuecomment-808338996 does not probably make sense.

**Comment by owais** [comment]
@pellared updated. PTAL

**Review by pellared — APPROVED** [review]
Please add description to the PR

**Pull request merged**